src/server/engine_shard.cc

```diff
@@ -538,8 +538,9 @@ void EngineShard::StartPeriodicHeartbeatFiber(util::ProactorBase* pb) {
 
   std::chrono::milliseconds period_ms(*cycle_ms);
 
+  fb2::Fiber::Opts fb_opts{.priority = fb2::FiberPriority::BACKGROUND, .name = "heatbeat"};
   fiber_heartbeat_periodic_ =
-      MakeFiber([this, index = pb->GetPoolIndex(), period_ms, heartbeat]() mutable {
+      fb2::Fiber(fb_opts, [this, index = pb->GetPoolIndex(), period_ms, heartbeat]() mutable {
```

Comment:
1. Heartbeat should be background
2. If its background, it doesn't penalize other background fibers on idle
3. We can split it into stages to increase simplicity and have dynamic load on it